src/Common/PoolBase.h

```diff
@@ -144,12 +144,17 @@ class PoolBase : private boost::noncopyable
                 return Entry(*items.back());
             }
 
-            LOG_INFO(log, "No free connections in pool. Waiting.");
-
             if (timeout < 0)
+            {
```

Comment:
This seems like it should log before calling wait (or wait_for in the else), not after.

Reply:
Nice catch, indeed, thanks, fixed.